Documentation/project-docs/testing.md

```diff
@@ -0,0 +1,74 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
+We usually run tests in parallel, unless we add an exception to the root of the project. Therefore we need to be careful to avoid possible side-effects when manipulating static members (e.g. properties).
```

Comment:
test fixtures are run in parallel with each other, but not test methods. 
another option is marking with attributes to reduce parallelism, you might mention that.